dcpy/models/base.py

```diff
@@ -131,3 +132,119 @@ def from_yaml(cls, yaml_str: str, *, template_vars=None) -> typing.Self:
     def from_path(cls, path: Path, *, template_vars=None) -> typing.Self:
         with open(path, "r", encoding="utf-8") as raw:
             return cls.from_yaml(raw.read(), template_vars=template_vars)
+
+
+class IndentedPrint(BaseModel, arbitrary_types_allowed=True):
```

Comment:
So it seems like this would be useful for printing any sort of dictionary. So I wonder if it would make sense to move the meat of this to `dcpy.utils.collections`, and operate on a dictionary. 

Then for `_display_names`, it looks like pydantic has a [serialization_alias](https://docs.pydantic.dev/latest/concepts/alias/) for the Field object. Haven't used/tested it, but if that approach would be a little cleaner, and more broadly usable?

Reply:
Also, does this deal at all with recursive references in pydantic objects?

Reply:
Can you clarify what you mean by dealing with recursive references?

Reply:
The nice thing about having it as a class and relying on nested objects also being IndentedPrint is that then you can specify these configs at different levels. 

Though looking at the configs, "_title", "_max_df_length", and "_pretty_print_fields" all seem like they'd really be fine being specified at time of function call. That leaves _display_names, which I agree with your suggestion, and then just the line_breaks - which couldn't be done at multiple levels but maybe can just be optional at time of function call at the top level (seems like it'd get cluttered if you had them at multiple levels anyways.

So sounds good to me! Will refactor

Reply:
> Can you clarify what you mean by dealing with recursive references?

Yeah, sorry, that was not clear at all. If I have ModelA, which refers to ModelB, which refers to ModelA, etc. So really circular references. I know pydantic is clever about how it serializes those, but I was just wondering if there's a potential infinite loop somewhere here.

Reply:
Well this will be a moot point after this is a util that assumes you've model dumped already, but I don't think so - it just goes into each node and assuming each node is a `IndentedPrint`, it calls that instances function to flatten/indent it